S2VX.Game/Editor/ToolState/SelectToolState.cs

```diff
@@ -173,26 +177,27 @@ public class SelectToolState : S2VXToolState {
             return gameTime + (leftOffset <= rightOffset ? -leftOffset : rightOffset);
         }
 
+        private double GetGameTimeAtMouse(Vector2 localMousePos) {
+            var tickBarNoteSelections = Editor.NotesTimeline.TickBarNoteSelections;
+            var mousePosX = ToSpaceOfOtherDrawable(ToLocalSpace(localMousePos), tickBarNoteSelections).X;
+            // temp until NoteTimeline Scroll on drag is implemented
+            mousePosX = Math.Clamp(mousePosX, 0, tickBarNoteSelections.DrawWidth);
+            var relativeMousePosX = mousePosX / tickBarNoteSelections.DrawWidth;
+            var gameTimeDeltaFromMiddle = (relativeMousePosX - 0.5f) * Editor.NotesTimeline.SectionLength * NotesTimeline.SecondsToMS;
+            return Time.Current + gameTimeDeltaFromMiddle;
+        }
+
         public override void OnToolDrag(DragEvent e) {
             if (!DelayDrag) {
                 switch (ToDrag) {
                     case SelectToolDragState.DragTimelineNote: {
-                        var tickBarNoteSelections = Editor.NotesTimeline.TickBarNoteSelections;
-                        var mousePosX = ToSpaceOfOtherDrawable(ToLocalSpace(e.ScreenSpaceMousePosition), tickBarNoteSelections).X;
-                        // temp until NoteTimeline Scroll on drag is implemented
-                        mousePosX = Math.Clamp(mousePosX, 0, tickBarNoteSelections.DrawWidth);
-                        var relativeMousePosX = mousePosX / tickBarNoteSelections.DrawWidth;
-                        var gameTimeDeltaFromMiddle = (relativeMousePosX - 0.5f) * Editor.NotesTimeline.SectionLength * NotesTimeline.SecondsToMS;
-                        var gameTimeAtMouse = Time.Current + gameTimeDeltaFromMiddle;
-
-                        var selectedNoteToTimeCopy = new Dictionary<Note, double>(SelectedNoteToTime);
-                        foreach (var noteAndTime in SelectedNoteToTime) {
-                            var note = noteAndTime.Key;
+                        var gameTimeAtMouse = GetGameTimeAtMouse(e.ScreenSpaceMousePosition);
+
+                        foreach (var note in SelectedNoteToTime.Keys.ToList()) {
```

Comment:
Don't think you need ToList(), SelectedNoteToTime.Keys is an IEnumerable, and you can enumerate through this without having to turn it into a list. Should only convert it to List if you need more List functionalities.

Reply:
I'm using .ToList() to make a copy of the keys so it doesn't throw an error because I'm editing the dictionary I'm iterating over. I was using a temp dict to do this before but I changed it to try to get something else working

Reply:
That's right I remember that, sounds good